### PR TITLE
Support for Dictionary and Dictionary Item APIs

### PR DESCRIPTION
**Pull request by Lizards**
This provides Models and API client methods for working with Dictionary and Dictionary Item APIs.

### CODE REVIEW

**Comment by axblum** [comment]
@Lizards, Have you tried to talk to Fastly Support on Merging this PR?

**Comment by psbanka** [comment]
@Lizards — can you fix the merge conflicts? If so, we can get this thing merged.

File: fastly/models.py [hunk 1]
```diff
@@ -206,3 +206,47 @@ def download(self):
...
+        params_str = urlencode(instance.attrs)
+        resp, data = instance._query('PUT', body=params_str)
+        instance._original_attrs = data
+        instance.attrs = data
+        return instance
+
+    def _collection_query(self, method, suffix='', body=None):
+        # DictionaryItem's create API URL is inconsistent with the other APIs,
+        # with the use of singular and plural ("item" and "items").
+        # Only use case for POST on this Model is to create one item, so swap in that API URL.
```
Comment (jevansnyc) [inline fastly/models.py]: its not inconsistent necessarily. "items" was intended to be used with PATCH for bulk updates.

**Comment by Lizards** [comment]
> @Lizards — can you fix the merge conflicts? If so, we can get this thing merged.

Resolved.  Thanks for taking a look at this!  :cool:

**Pull request merged**